lib/rspec/matchers/built_in/compound.rb

```diff
@@ -155,6 +155,13 @@ def initialize(actual, matcher_1, matcher_2)
 
           def matcher_matches?(matcher)
             @match_results.fetch(matcher)
+          rescue IndexError
+            raise ArgumentError,
+              """
+                A matcher has no match results, this can occur when an unexpected
+                call stack or local jump occurs. Prehaps one of your matchers needs
+                to declare `expects_call_stack_jump?` as true?
+              """.gsub(/^\s+/,'').tr("\n"," ")
```

Comment:
Haha, I just realized you are removing the wrapping here....I hadn't noticed.  I still think `\` is simpler.